### ISSUE TITLE
tracer extension build broken (master) (windows)

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#449]
*Originally reported by* **Anthony Sottile (Bitbucket: [asottile](https://bitbucket.org/asottile), GitHub: [asottile](https://github.com/asottile))**

----------------------------------------

Finally getting around to verifying the fix for #420, but it seems master's build is broken on windows:

Important bits:

```
C:\Users\Anthony\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python27\include -Ic:\users\anthony\desktop\git\coveragepy\venv\PC /Tccoverage/ctracer/datastack.c /Fobuild\temp.win32-2.7\Release\coverage/ctracer/datastack.obj
  datastack.c
  coverage/ctracer/datastack.c(31) : error C2143: syntax error : missing ';' before 'type'
  coverage/ctracer/datastack.c(32) : error C2275: 'DataStackEntry' : illegal use of this type as an expression
          c:\temp\pip-rl6no4-build\coverage\ctracer\datastack.h(32) : see declaration of 'DataStackEntry'
  coverage/ctracer/datastack.c(32) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(32) : error C2065: 'bigger' : undeclared identifier
  coverage/ctracer/datastack.c(33) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(33) : warning C4047: '==' : 'int' differs in levels of indirection from 'void *'
  coverage/ctracer/datastack.c(38) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(38) : warning C4047: '=' : 'DataStackEntry *' differs in levels of indirection from 'int'
  coverage/ctracer/datastack.c(39) : error C2065: 'bigger' : undeclared identifier
  **
  ** Couldn't install with extension module, trying without it...
  ** BuildFailed: command 'C:\\Users\\Anthony\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\cl.exe' failed with exit status 2
  **
```

Full output:

```
$ pip wheel . -v
Processing c:\users\anthony\desktop\git\coveragepy
  Running setup.py (path:c:\temp\pip-rl6no4-build\setup.py) egg_info for package from file:///C:/Users/Anthony/Desktop/git/coveragepy
    Running command python setup.py egg_info
    running egg_info
    creating pip-egg-info\coverage.egg-info
    writing pip-egg-info\coverage.egg-info\PKG-INFO
    writing top-level names to pip-egg-info\coverage.egg-info\top_level.txt
    writing dependency_links to pip-egg-info\coverage.egg-info\dependency_links.txt
    writing entry points to pip-egg-info\coverage.egg-info\entry_points.txt
    writing manifest file 'pip-egg-info\coverage.egg-info\SOURCES.txt'
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'pip-egg-info\coverage.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files found matching 'ci\appveyor.token'
    no previously-included directories found matching 'doc\_build'
    no previously-included directories found matching 'tests\eggsrc\build'
    no previously-included directories found matching 'tests\eggsrc\dist'
    no previously-included directories found matching 'tests\eggsrc\*.egg-info'
    warning: no previously-included files matching '*.py[co]' found anywhere in distribution
    writing manifest file 'pip-egg-info\coverage.egg-info\SOURCES.txt'
  Source in c:\temp\pip-rl6no4-build has version 4.1a0, which satisfies requirement coverage==4.1a0 from file:///C:/Users/Anthony/Desktop/git/coveragepy
Building wheels for collected packages: coverage
  Running setup.py bdist_wheel for coverage
  Destination directory: c:\temp\tmprnn3cypip-wheel-
  Running command c:\users\anthony\desktop\git\coveragepy\venv\scripts\python.exe -c "import setuptools;__file__='c:\\temp\\pip-rl6no4-build\\setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d c:\temp\tmprnn3cypip-wheel-
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win32-2.7
  creating build\lib.win32-2.7\coverage
  copying coverage\annotate.py -> build\lib.win32-2.7\coverage
  copying coverage\backunittest.py -> build\lib.win32-2.7\coverage
  copying coverage\backward.py -> build\lib.win32-2.7\coverage
  copying coverage\bytecode.py -> build\lib.win32-2.7\coverage
  copying coverage\cmdline.py -> build\lib.win32-2.7\coverage
  copying coverage\collector.py -> build\lib.win32-2.7\coverage
  copying coverage\config.py -> build\lib.win32-2.7\coverage
  copying coverage\control.py -> build\lib.win32-2.7\coverage
  copying coverage\data.py -> build\lib.win32-2.7\coverage
  copying coverage\debug.py -> build\lib.win32-2.7\coverage
  copying coverage\env.py -> build\lib.win32-2.7\coverage
  copying coverage\execfile.py -> build\lib.win32-2.7\coverage
  copying coverage\files.py -> build\lib.win32-2.7\coverage
  copying coverage\html.py -> build\lib.win32-2.7\coverage
  copying coverage\misc.py -> build\lib.win32-2.7\coverage
  copying coverage\monkey.py -> build\lib.win32-2.7\coverage
  copying coverage\parser.py -> build\lib.win32-2.7\coverage
  copying coverage\phystokens.py -> build\lib.win32-2.7\coverage
  copying coverage\pickle2json.py -> build\lib.win32-2.7\coverage
  copying coverage\plugin.py -> build\lib.win32-2.7\coverage
  copying coverage\plugin_support.py -> build\lib.win32-2.7\coverage
  copying coverage\python.py -> build\lib.win32-2.7\coverage
  copying coverage\pytracer.py -> build\lib.win32-2.7\coverage
  copying coverage\report.py -> build\lib.win32-2.7\coverage
  copying coverage\results.py -> build\lib.win32-2.7\coverage
  copying coverage\summary.py -> build\lib.win32-2.7\coverage
  copying coverage\templite.py -> build\lib.win32-2.7\coverage
  copying coverage\test_helpers.py -> build\lib.win32-2.7\coverage
  copying coverage\version.py -> build\lib.win32-2.7\coverage
  copying coverage\xmlreport.py -> build\lib.win32-2.7\coverage
  copying coverage\__init__.py -> build\lib.win32-2.7\coverage
  copying coverage\__main__.py -> build\lib.win32-2.7\coverage
  creating build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\coverage_html.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\index.html -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.debounce.min.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.hotkeys.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.isonscreen.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.min.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.tablesorter.min.js -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\keybd_closed.png -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\keybd_open.png -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\pyfile.html -> build\lib.win32-2.7\coverage\htmlfiles
  copying coverage\htmlfiles\style.css -> build\lib.win32-2.7\coverage\htmlfiles
  running build_ext
  building 'coverage.tracer' extension
  creating build\temp.win32-2.7
  creating build\temp.win32-2.7\Release
  creating build\temp.win32-2.7\Release\coverage
  creating build\temp.win32-2.7\Release\coverage\ctracer
  C:\Users\Anthony\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python27\include -Ic:\users\anthony\desktop\git\coveragepy\venv\PC /Tccoverage/ctracer/datastack.c /Fobuild\temp.win32-2.7\Release\coverage/ctracer/datastack.obj
  datastack.c
  coverage/ctracer/datastack.c(31) : error C2143: syntax error : missing ';' before 'type'
  coverage/ctracer/datastack.c(32) : error C2275: 'DataStackEntry' : illegal use of this type as an expression
          c:\temp\pip-rl6no4-build\coverage\ctracer\datastack.h(32) : see declaration of 'DataStackEntry'
  coverage/ctracer/datastack.c(32) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(32) : error C2065: 'bigger' : undeclared identifier
  coverage/ctracer/datastack.c(33) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(33) : warning C4047: '==' : 'int' differs in levels of indirection from 'void *'
  coverage/ctracer/datastack.c(38) : error C2065: 'bigger_data_stack' : undeclared identifier
  coverage/ctracer/datastack.c(38) : warning C4047: '=' : 'DataStackEntry *' differs in levels of indirection from 'int'
  coverage/ctracer/datastack.c(39) : error C2065: 'bigger' : undeclared identifier
  **
  ** Couldn't install with extension module, trying without it...
  ** BuildFailed: command 'C:\\Users\\Anthony\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\cl.exe' failed with exit status 2
  **
  running bdist_wheel
  running build
  running build_py
  creating build\lib
  creating build\lib\coverage
  copying coverage\annotate.py -> build\lib\coverage
  copying coverage\backunittest.py -> build\lib\coverage
  copying coverage\backward.py -> build\lib\coverage
  copying coverage\bytecode.py -> build\lib\coverage
  copying coverage\cmdline.py -> build\lib\coverage
  copying coverage\collector.py -> build\lib\coverage
  copying coverage\config.py -> build\lib\coverage
  copying coverage\control.py -> build\lib\coverage
  copying coverage\data.py -> build\lib\coverage
  copying coverage\debug.py -> build\lib\coverage
  copying coverage\env.py -> build\lib\coverage
  copying coverage\execfile.py -> build\lib\coverage
  copying coverage\files.py -> build\lib\coverage
  copying coverage\html.py -> build\lib\coverage
  copying coverage\misc.py -> build\lib\coverage
  copying coverage\monkey.py -> build\lib\coverage
  copying coverage\parser.py -> build\lib\coverage
  copying coverage\phystokens.py -> build\lib\coverage
  copying coverage\pickle2json.py -> build\lib\coverage
  copying coverage\plugin.py -> build\lib\coverage
  copying coverage\plugin_support.py -> build\lib\coverage
  copying coverage\python.py -> build\lib\coverage
  copying coverage\pytracer.py -> build\lib\coverage
  copying coverage\report.py -> build\lib\coverage
  copying coverage\results.py -> build\lib\coverage
  copying coverage\summary.py -> build\lib\coverage
  copying coverage\templite.py -> build\lib\coverage
  copying coverage\test_helpers.py -> build\lib\coverage
  copying coverage\version.py -> build\lib\coverage
  copying coverage\xmlreport.py -> build\lib\coverage
  copying coverage\__init__.py -> build\lib\coverage
  copying coverage\__main__.py -> build\lib\coverage
  creating build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\coverage_html.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\index.html -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.debounce.min.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.hotkeys.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.isonscreen.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.min.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\jquery.tablesorter.min.js -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\keybd_closed.png -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\keybd_open.png -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\pyfile.html -> build\lib\coverage\htmlfiles
  copying coverage\htmlfiles\style.css -> build\lib\coverage\htmlfiles
  installing to build\bdist.win32\wheel
  running install
  running install_lib
  creating build\bdist.win32
  creating build\bdist.win32\wheel
  creating build\bdist.win32\wheel\coverage
  copying build\lib\coverage\annotate.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\backunittest.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\backward.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\bytecode.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\cmdline.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\collector.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\config.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\control.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\data.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\debug.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\env.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\execfile.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\files.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\html.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  creating build\bdist.win32\wheel\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\coverage_html.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\index.html -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\jquery.debounce.min.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\jquery.hotkeys.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\jquery.isonscreen.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\jquery.min.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\jquery.tablesorter.min.js -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\keybd_closed.png -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\keybd_open.png -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\pyfile.html -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\htmlfiles\style.css -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage\htmlfiles
  copying build\lib\coverage\misc.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\monkey.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\parser.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\phystokens.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\pickle2json.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\plugin.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\plugin_support.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\python.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\pytracer.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\report.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\results.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\summary.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\templite.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\test_helpers.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\version.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\xmlreport.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\__init__.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  copying build\lib\coverage\__main__.py -> build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage
  running install_egg_info
  running egg_info
  creating coverage.egg-info
  writing coverage.egg-info\PKG-INFO
  writing top-level names to coverage.egg-info\top_level.txt
  writing dependency_links to coverage.egg-info\dependency_links.txt
  writing entry points to coverage.egg-info\entry_points.txt
  writing manifest file 'coverage.egg-info\SOURCES.txt'
  warning: manifest_maker: standard file '-c' not found

  reading manifest file 'coverage.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no previously-included files found matching 'ci\appveyor.token'
  no previously-included directories found matching 'doc\_build'
  no previously-included directories found matching 'tests\eggsrc\build'
  no previously-included directories found matching 'tests\eggsrc\dist'
  no previously-included directories found matching 'tests\eggsrc\*.egg-info'
  warning: no previously-included files matching '*.py[co]' found anywhere in distribution
  writing manifest file 'coverage.egg-info\SOURCES.txt'
  Copying coverage.egg-info to build\bdist.win32\wheel\coverage-4.1a0.data\..\coverage-4.1a0-py2.7.egg-info
  running install_scripts
  creating build\bdist.win32\wheel\coverage-4.1a0.dist-info\WHEEL
  Stored in directory: c:\users\anthony\desktop\git\coveragepy\wheelhouse
Successfully built coverage
Cleaning up...
  Removing source in c:\temp\pip-rl6no4-build
```

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/449

**Comment by nedbat** [comment]
It looks to me like you are using the code from GitHub's master.  Currently, that is stuck on a commit from the who-tests-what-170 branch.  Notice that it claims to have 4.1a0.  I'm trying to get the GitHub mirror fixed, but in the meantime, use Bitbucket as the source.

The compile error is a real problem, but only when STATS is turned on.  I'll fix that.

**Comment by nedbat** [comment]
Compile error fixed in [7ee1f28e65e8 (bb)](https://bitbucket.org/ned/coveragepy/commits/7ee1f28e65e8)
